static/js/upload_widget.ts

```diff
@@ -45,11 +134,30 @@ export function build_widget(
         $file_name_field.text("");
         $clear_button.hide();
         $upload_button.show();
-        if ($preview_text !== undefined) {
-            $preview_text.hide();
+        $select_button?.hide();
+        $(
+            "#add-custom-emoji-modal .dialog_submit_button, #add-custom-emoji-modal .dialog_cancel_button,.emoji_name_input",
+        ).show();
+        if ($preview_text !== undefined && $select_button) {
+            uppy.close();
         }
     }
 
+    $select_button?.on("click", (e) => {
+        const file = uppy.getFiles()[0];
+        const url = URL.createObjectURL(file.data);
+        $preview_image?.attr("src", url);
+        cropfile = new File([file.data], file.name);
+        uppy.close();
+        $select_button?.hide();
+        $clear_button.hide();
+        $(
+            "#add-custom-emoji-modal .dialog_submit_button, #add-custom-emoji-modal .dialog_cancel_button,.emoji_name_input",
+        ).show();
```

Comment:
This selector is pretty weird -- we should probably want to compute what to show from which element we are.

(And I don't think we want to hardcode this image cropping feature to just work with the custom emoji input -- we should make it an optional setting for the upload widget).

Reply:
Yes, I am aware that hardcoding the features is not recommended, and I will make sure to implement them as soft-coded when working on the remaining part of the issue, which includes adding the same feature to profile pictures and organization logo uploads.